macos/FluentUI/Button.swift

```diff
@@ -0,0 +1,230 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import AppKit
+
+/// Indicates what style our button is drawn as
+@objc public enum ButtonStyle: Int, CaseIterable {
+    case primaryFilled	// Solid fill color
+    case primaryOutline	// Clear fill color, solid outline
+    case borderless		// Clear fill color, clear outline
+}
+
+// MARK: - Button
+
+/// A fluent styled button, with hover effects and a corner radius.
+@objc(MSFButton)
+open class Button: NSButton {
+
+	@objc public init(title: String, style: ButtonStyle = .primaryFilled) {
+		super.init(frame: .zero)
+		self.title = title
+		self.style = style
+		initialize()
+    }
+	
+	@objc public init(image: NSImage, style: ButtonStyle = .primaryFilled) {
+		super.init(frame: .zero)
+		self.style = style
+		initialize()
+		// Image must be set after `wantsLayer = true` so that the image is on the layer and not the underlying NSButtonCell
+		imagePosition = .imageOnly
+		self.image = image
+    }    
+	
+	public override init(frame frameRect: NSRect) {
+		super.init(frame: frameRect)
+		initialize()
+	}
+
+	@available(*, unavailable)
+	required public init?(coder decoder: NSCoder) {
+		preconditionFailure()
+	}
+	
+	private func initialize() {
+		// Do common initialization work
+		isBordered = false
+		wantsLayer = true
+		update()
+	}
+	
+	override public var image: NSImage? {
+		didSet {
+			if imagePosition != .imageOnly {
+				preconditionFailure("We currently do not support adding both an image and a title. Please set the imagePosition to .imageOnly before setting an image")
+			}
+		}
+	}
+	
+	override public func updateTrackingAreas() {
+		super.updateTrackingAreas()
+		
+		// Remove existing trackingArea
+		if let trackingArea = trackingArea {
+			removeTrackingArea(trackingArea)
+			self.trackingArea = nil
+		}
+		
+		// Create a new trackingArea
+		let opts: NSTrackingArea.Options = [.mouseEnteredAndExited, .activeAlways]
+		let trackingArea = NSTrackingArea(
+			rect: bounds,
+			options: opts,
+			owner: self,
+			userInfo: nil
+		)
+		addTrackingArea(trackingArea)
+		self.trackingArea = trackingArea
+	}
+	
+	override public func mouseEntered(with event: NSEvent) {
```

Comment:
open override? Does it actually need to be public?

Reply:
No, but I thought we defaulted to public unless otherwise needed?

Reply:
I would default the opposite - only expose things for a reason. I don't know if it's verifiable fact that not making things public reduces size, but that makes sense in my head - I'd assume it takes extra instructions/code to expose something publicly